block/internal/syncing/p2p_handler.go

```diff
@@ -50,7 +55,7 @@ func NewP2PHandler(
 
 // SetProcessedHeight updates the highest processed block height.
 func (h *P2PHandler) SetProcessedHeight(height uint64) {
-	for {
+	for range 1_000 {
```

Comment:
ref: https://github.com/evstack/ev-node/pull/2868/files#diff-b4e2c94aa607c5394b5031021292365f74597ac046d65186e15980b6a3e2f9dcR105 / https://github.com/evstack/ev-node/pull/2873#discussion_r2549079388 to keep consistency